components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java

```diff
@@ -148,7 +148,11 @@ public void buildResourceAccessControlData() {
             List<AuthenticationHandler> allAvailableAuthHandlers =
                     AuthenticationServiceHolder.getInstance().getAuthenticationHandlers();
             for (AuthenticationHandler handler : allAvailableAuthHandlers) {
-                allowedAuthHandlersList.add(handler.getName());
+                String handlerName = handler.getName();
+                if (handlerName.equals(Constants.BASIC_CLIENT_AUTH_HANDLER)) {
```

Comment:
use Constants.BASIC_CLIENT_AUTH_HANDLER.equals(handlerName) to avoid possible NPEs